### PR TITLE
[PREVIEW] All_CHANGES

### PR DESCRIPTION
**Pull request by schernolyas**
Changes: 1) Merge with branch 'origin/support_basic_types' 2) rework exceptions 3) refactoring

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -1,194 +1,193 @@
...
+			<artifactId>jboss-transaction-api_1.2_spec</artifactId>
+			<scope>provided</scope>
+			<!-- I do hope it is optional but that needs to be verified -->
+			<optional>true</optional>
+		</dependency>
+		<dependency>
+			<groupId>org.javassist</groupId>
+			<artifactId>javassist</artifactId>
+		</dependency>
+		<dependency>
```
Comment (DavideD) [inline core/pom.xml]: Please, start using our templates or we will have thousands of changes like this on the diff

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java [hunk 2]
```diff
@@ -63,123 +97,316 @@ public OrientDBDialect(OrientDBDatastoreProvider provider) {
...
-		throw new UnsupportedOperationException( "Not supported yet." );
+
+		log.debug( "createAssociation: AssociationKey:" + key + "; AssociationContext" + associationContext );
+		return new Association();
 	}
 
 	@Override
 	public void insertOrUpdateAssociation(AssociationKey key, Association association, AssociationContext associationContext) {
-		throw new UnsupportedOperationException( "Not supported yet." );
+		log.debug( "insertOrUpdateAssociation: AssociationKey:" + key + "; AssociationContext:" + associationContext + "; association:" + association );
```
Comment (DavideD) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java]: I would prefere to have an exception saying that this method is not implemented than a log

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java [hunk 3]
```diff
@@ -189,18 +416,93 @@ public boolean isStoredInEntityStructure(AssociationKeyMetadata associationKeyMe
...
+	private String buildNativeQuery(BackendQuery<String> customQuery, QueryParameters queryParameters) {
+		return customQuery.getQuery();
+	}
+
+	/**
+	 * Returns a map with the named parameter values from the given parameters object, converted by the {@link GridType}
+	 * corresponding to each parameter type.
+	 */
+	private Map<String, Object> getNamedParameterValuesConvertedByGridType(QueryParameters queryParameters) {
+		log.debug( "getNamedParameterValuesConvertedByGridType. named parameters: " + queryParameters.getNamedParameters().size() );
```
Comment (DavideD) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java]: It would be better to use the debugf for these logs, this way you can aviod string concatanation when the log is not required

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java [hunk 4]
```diff
@@ -251,13 +562,49 @@ public void sessionFactoryCreated(SessionFactoryImplementor sessionFactoryImplem
 
 	@Override
 	public GridType overrideType(Type type) {
-		log.info( "overrideType:" + type.getName() + ";" + type.getReturnedClass() );
+		log.debug( "overrideType:" + type.getName() + ";" + type.getReturnedClass() + ";" );
 		GridType gridType = null;
-		if ( type.getName().equals( "com.orientechnologies.orient.core.id.ORecordId" ) ) {
+
+		if ( type.getName().equals( ORecordId.class.getName() ) ) {
```
Comment (DavideD) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java]: ca nyou use instanceof to check this classes?

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/logging/impl/Log.java [hunk 5]
```diff
@@ -52,4 +43,37 @@
 
 	@Message(id = 1408, value = "Error while cheking transaction status")
 	HibernateException exceptionWhileChekingTransactionStatus(@Cause Exception e);
+
+	@Message(id = 1700, value = "Cannot create class %s")
+	HibernateException cannotGenerateVertexClass(String className, @Cause Exception cause);
```
Comment (DavideD) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/logging/impl/Log.java]: All the id in this class should be in the same range, could you update the existing one as well to 17xx instead of 14xx

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/query/parsing/impl/OrientDBBasedQueryParserService.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import org.hibernate.ogm.query.spi.BaseQueryParserService;
+import org.hibernate.ogm.query.spi.QueryParsingResult;
+import org.hibernate.ogm.service.impl.SessionFactoryEntityNamesResolver;
+import org.hibernate.ogm.util.impl.Log;
+import org.hibernate.ogm.util.impl.LoggerFactory;
+
+/**
+ * @author chernolyassv
+ */
+public class OrientDBBasedQueryParserService extends BaseQueryParserService {
```
Comment (DavideD) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/query/parsing/impl/OrientDBBasedQueryParserService.java]: This class does not do anything, can we remove it for now?

File: orientdb/src/test/java/org/hibernate/datastore/ogm/orientdb/utils/OrientDBBackendTckHelper.java [hunk 7]
```diff
@@ -6,18 +6,30 @@
...
 public class OrientDBBackendTckHelper {
+
+	@BeforeClass
+	public static void setUpClass() {
+		MemoryDBUtil.createDbFactory( MEMORY_TEST );
+	}
+
+	@AfterClass
+	public static void tearDownClass() {
+		MemoryDBUtil.recrateInMemoryDn( MEMORY_TEST );
```
Comment (DavideD) [inline orientdb/src/test/java/org/hibernate/datastore/ogm/orientdb/utils/OrientDBBackendTckHelper.java]: What's happening here? Are you testing something here?

File: pom.xml [hunk 8]
```diff
@@ -378,6 +378,7 @@
                         <systemPropertyVariables>
                            <!-- See test org.hibernate.ogm.test.utils.EnvironmentTest -->
                            <hibernate.service.allow_crawling>false</hibernate.service.allow_crawling>
+                           <user.timezone>UTC</user.timezone>
```
Comment (DavideD) [inline pom.xml]: Why do you need this? Also, this is an example of a change that should be in a separate commit

File: pom.xml [hunk 9]
```diff
@@ -24,16 +24,16 @@
...
         <module>neo4j</module>
-        <module>couchdb</module>
+        <module>orientdb</module>
+       <module>couchdb</module>
         <module>cassandra</module>
         <module>redis</module>
-        <module>orientdb</module>
         <module>modules</module>
-        <module>performance</module>
+        <module>performance</module>
```
Comment (DavideD) [inline pom.xml]: How come this file is changed?

File: bom/pom.xml [hunk 10]
```diff
@@ -27,7 +27,7 @@
         <infinispanVersion>8.0.2.Final</infinispanVersion>
         <mongodbVersion>3.0.3</mongodbVersion>
         <fongodbVersion>2.0.2</fongodbVersion>
-        <orientdbVersion>2.1.9</orientdbVersion>
+        <orientdbVersion>[2.1.12,)</orientdbVersion>
```
Comment (DavideD) [inline bom/pom.xml]: As a reminder: we don't use range version, use 2.1.12
This should be a separate comimt.

File: core/src/test/java/org/hibernate/ogm/backendtck/optimisticlocking/OptimisticLockingTest.java [hunk 11]
```diff
@@ -13,7 +13,7 @@
 import static org.hibernate.ogm.utils.GridDialectType.HASHMAP;
 import static org.hibernate.ogm.utils.GridDialectType.INFINISPAN;
 import static org.hibernate.ogm.utils.GridDialectType.NEO4J;
-import static org.hibernate.ogm.utils.GridDialectType.REDIS_JSON;
+import static org.hibernate.ogm.utils.GridDialectType.REDIS;
```
Comment (DavideD) [inline core/src/test/java/org/hibernate/ogm/backendtck/optimisticlocking/OptimisticLockingTest.java]: Did you change this?

**Comment by DavideD** [comment]
Could you add a description of what this pull request is supposed to solve?

I understand that this is a preview but could you make sure to remove all the unrelated changes next time? This pull request contains changes to REDIS that should not be there.

Is there test that I can try?

File: bom/pom.xml (hunk 10 above)
Comment (DavideD) [inline bom/pom.xml]: As a reminder, we don't use range, just leave 2.1.12
And if you can put it in a sperate commit I can start include something

Comment (schernolyas) [inline bom/pom.xml]: fixed

File: pom.xml (hunk 8 above)
Comment (schernolyas) [inline pom.xml]: fixed

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/query/parsing/impl/OrientDBBasedQueryParserService.java (hunk 6 above)
Comment (schernolyas) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/query/parsing/impl/OrientDBBasedQueryParserService.java]: fixed

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/logging/impl/Log.java (hunk 5 above)
Comment (schernolyas) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/logging/impl/Log.java]: fixed

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java (hunk 2 above)
Comment (schernolyas) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java]: fixed

**Comment by schernolyas** [comment]
closed
